src/Tizen.NUI/src/public/BaseComponents/ControlState.cs

```diff
@@ -0,0 +1,315 @@
+
+            state = new ControlState(name);
+            state.stateList.Add(state);
+            stateDictionary.Add(name, state);
+            return state;
+        }
+
+        /// <summary>
+        /// Create an instance of the <see cref="ControlState"/> with combined states.
+        /// </summary>
+        /// <param name="states">The control state array.</param>
+        /// <returns>The <see cref="ControlState"/> instance which has combined state.</returns>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static ControlState Create(params ControlState[] states)
+        {
+            if (states.Length == 1)
+                return states[0];
+
+            ControlState newState = new ControlState();
+            for (int i = 0; i < states.Length; i++)
+            {
+                if (states[i] == Normal)
+                    continue;
+
+                if (states[i] == All)
+                    return All;
+
+                newState.stateList.AddRange(states[i].stateList);
+            }
+
+            if (newState.stateList.Count == 0)
+                return Normal;
+
+            newState.stateList = newState.stateList.Distinct().ToList();
+
+            return newState;
+        }
+
+        /// <summary>
+        /// Determines whether a state contains a specified state.
+        /// </summary>
+        /// <param name="state">The state to search for</param>
+        /// <returns>true if the state contain a specified state, otherwise, false.</returns>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public bool Contains(ControlState state)
+        {
+            if (state == null)
+                throw new ArgumentNullException(nameof(state));
+
+            if (!IsCombined)
+                return ReferenceEquals(this, state);
+
+            bool found;
+            for (int i = 0; i < state.stateList.Count; i++)
+            {
+                found = false;
+                for (int j = 0; j < stateList.Count; j++)
+                {
+                    if (ReferenceEquals(state.stateList[i], stateList[j]))
+                    {
+                        found = true;
+                        break;
+                    }
+                }
+                if (!found) return false;
+            }
+
+            return true;
+        }
+
+        ///  <inheritdoc/>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public bool Equals(ControlState other)
+        {
+            if (other is null || stateList.Count != other.stateList.Count)
+                return false;
+
+            return Contains(other);
+        }
+
+        ///  <inheritdoc/>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public override bool Equals(object obj) => this.Equals(obj as ControlState);
+
+        ///  <inheritdoc/>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public override int GetHashCode() => stateList.GetHashCode();
```

Comment:
Is it safe? I think List.GetHashCode() may use stateList[i].GetHashCode() which can raises infinite loop.